pkgs/development/libraries/gobject-introspection/wrapper.nix

```diff
@@ -30,13 +30,67 @@ in
 # wrap both pkgsCrossX.buildPackages.gobject-introspection and {pkgs,pkgsSomethingExecutableOnBuildSystem).buildPackages.gobject-introspection
 if (!stdenv.hostPlatform.canExecute stdenv.targetPlatform) && stdenv.targetPlatform.emulatorAvailable buildPackages
 then
-  stdenv.mkDerivation
-    (builtins.removeAttrs overridenUnwrappedGir.drvAttrs [ "name" ] # so we can get a fresh name generated from the pname
-      // {
+  overridenUnwrappedGir.overrideAttrs
+    (previousAttrs:
+      {
 
-      inherit (overridenUnwrappedGir) meta;
+        pname = "gobject-introspection-wrapped";
+        passthru = previousAttrs.passthru // {
+          unwrapped = overridenUnwrappedGir;
+        };
+        phases = [ "fixupPhase" ]; # don't remove, it is valid to set phases here.
```

Comment:
Can we not replace it with dontConfigure, dontBuild and dontInstall?

Reply:
switched to buildCommand and `eval fixupPhase`